Fix #128

lohedges opened the pull request:
This PR closes #128 by fixing the streaming operators for the `sire.vol.TriclinicBox` class. Since rotation and reduction operations are no longer automatically applied we need to add _all_ class member data to the binary data stream, not the minimal information from which the box can be recovered.

* I confirm that I have merged the latest version of `devel` into this branch before issuing this pull request (e.g. by running `git pull origin devel`): [y]
* I confirm that I have added a changelog entry to the changelog (we will add a link to this PR as part of the review): [y]
* I confirm that I have permission to release this code under the GPL3 license: [y]

## Suggested reviewers:
@chryswoods

chryswoods reviewed (changes requested):
I see the need for this, but we have to be careful changing the binary streaming format. This change changes the format (as it adds more things to be streamed for this object). We need to (1) change the stream version number from 1 to 2 to indicate the change in format and (2) preserve the old format on read so that we can still read old s3 files (e.g. read assuming that this hasn't been rotated)

lohedges commented:
Ah yes, good point. I'll also need to auto rotate and reduce if loading a v1 stream.

lohedges commented:
I've updated the version number and added support for loading versions 1 and 2. Weirdly, the system in [test_triclinic.py](https://github.com/OpenBioSim/sire/blob/4a1025e8de9b5fb97c2670fa1de68b6bbf3c9d9f/tests/vol/test_triclinic.py#L95-L104) _doesn't_ fail if I update the version number and _don't_ add support for the old version. This stream file was created quite a while ago so should surely have `v1` in the header :man_shrugging:

chryswoods reviewed (approved):
Strange that this still streamed the old s3 file without the fix.

This all looks good :-)
